CMake/external_fastdds.cmake

```diff
@@ -0,0 +1,84 @@
+if(BUILD_DDS_BACKEND)
+    include(ExternalProject)
+    
+    # Foonathan memory is a dependency of FastDDS.
+    ExternalProject_Add(
+        foonathan_memory
+        PREFIX foonathan_memory
+        GIT_REPOSITORY "https://github.com/foonathan/memory.git"
+        GIT_TAG "v0.7-1" # Commit Hash: 19ab0759c7f053d88657c0eb86d879493f784d61
```

Comment:
Where does this tag come from? FastDDS req? Ours? You might want to mention that in the comments

Reply:
It's the latest, no specific requirement from FastDDS.
I will add a comment

Reply:
But it means that we won't get newer tags if they're available, no?
Don't we want to get `master` or something?

Reply:
I don't like 'latest' approach.
We had issues with it in the past with other libraries we use when 1 day before a release they brake the API and the CI fails..
I preffet use a stable version an update on demand, when we read the release notes.